### PR TITLE
Token Metadata: refactor and add update_authority constraint

### PR DESCRIPTION
**Pull request by dankelleher**

### CODE REVIEW

File: programs/sunrise-stake/src/utils/metaplex.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        &[accounts.state.gsol_mint_authority_bump],
+    ];
+    let pda_signer = &[&seeds[..]];
+
+    let account_info = vec![
+        accounts.metadata.to_account_info(),
+        accounts.gsol_mint.to_account_info(),
+        accounts.gsol_mint_authority.to_account_info(),
+        accounts.update_authority.to_account_info(), // payer
+        accounts.update_authority.to_account_info(),
```
Comment (s6thgehr) [inline programs/sunrise-stake/src/utils/metaplex.rs:26]: This should be gsol_mint_authority then

**Review by s6thgehr — CHANGES_REQUESTED** [review]
Adapt the list of account_info

File: programs/sunrise-stake/src/utils/metaplex.rs (hunk 1 above)
Comment (dankelleher) [reply programs/sunrise-stake/src/utils/metaplex.rs]: good spot

**Review by s6thgehr — APPROVED** [review]

**Review by s6thgehr — APPROVED** [review]

**Review by s6thgehr — APPROVED** [review]
Looks good.

**Pull request merged**